dkc/core/models/folder.py

```diff
@@ -63,16 +63,12 @@ class Meta:
     def is_root(self) -> bool:
         return self.parent is None
 
-    def path_to_root(self) -> List[Folder]:
+    def iter_ancestors(self) -> Iterator[Folder]:
+        """Iterate up to the root of the Folder hierarchy, starting with this Folder."""
         folder = self
-        path = [folder]
-        while not folder.is_root:
+        while folder is not None:
+            yield folder
             folder = folder.parent
-            path.append(folder)
-            if len(path) > self.MAX_TREE_HEIGHT:
-                raise MaxFolderDepthExceeded()
```

Comment:
Note, this runtime check is removed, but should be unnecessary. We enforce `MAX_TREE_HEIGHT` in `_folder_pre_save`.